app/src/main/java/com/github/se/polyfit/ui/components/scaffold/CenteredTopBar.kt

```diff
@@ -28,7 +33,12 @@ fun CenteredTopBar(title: String, titleColor: Color = PrimaryPurple, navigateBac
       },
       navigationIcon = {
         IconButton(
-            onClick = { navigateBack() },
+            onClick = {
+              if (!isBackClicked) {
+                navigateBack()
+                isBackClicked = true
```

Comment:
```suggestion
                isBackClicked = true
                navigateBack()
```
just as a safety to ensure its blocked before a second function runs

Reply:
Ok thanks. I did not it could cause an issue. I will fix it right away !